src/containers/paint-editor-wrapper.jsx

```diff
@@ -22,28 +22,42 @@ class PaintEditorWrapper extends React.Component {
     handleUpdateName (name) {
         this.props.vm.renameCostume(this.props.selectedCostumeIndex, name);
     }
-    handleUpdateSvg (svg, rotationCenterX, rotationCenterY) {
-        this.props.vm.updateSvg(this.props.selectedCostumeIndex, svg, rotationCenterX, rotationCenterY);
+    handleUpdateImage (isVector, image, rotationCenterX, rotationCenterY) {
+        if (isVector) {
+            this.props.vm.updateSvg(
+                this.props.selectedCostumeIndex,
+                image,
+                rotationCenterX,
+                rotationCenterY);
+        } else {
+            this.props.vm.updateBitmap(
+                this.props.selectedCostumeIndex,
+                image,
+                rotationCenterX,
+                rotationCenterY,
+                2 /* bitmapResolution */);
```

Comment:
In the long term, do you imagine this `2` going away since that will be the only case at runtime?

Reply:
Yes, I think it would be good if one day Scratch 2 projects still using bitmap resolution 1 were converted by the 2.0-coping-strategies repo right away, and then the project json, gui, vm, paint editor, storage and renderer could all share the same definitions of size and rotation center and not store a resolution. Right now VM is in charge of dividing values by bitmap resolution before sending them to the renderer.